From 71902f2a11ad3a1e6bb6240593f8581464a3102a Mon Sep 17 00:00:00 2001
From: Wallace Wang <r59996@freescale.com>
Date: Wed, 19 Mar 2008 19:55:19 +0800
Subject: [PATCH] ENGR00069259 Add spdif to imx37 defconfig and fix an register setting

Add spdif driver to imx37 defconfig
For some imx37 boards, the first time to play pcm out from spdif fails

Signed-off-by: Wallace Wang <r59996@freescale.com>
---
 arch/arm/configs/imx37_3stack_defconfig |    2 +-
 sound/arm/mxc-alsa-spdif.c              |    2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/arch/arm/configs/imx37_3stack_defconfig b/arch/arm/configs/imx37_3stack_defconfig
index 8e2afb6..84a87e7 100644
--- a/arch/arm/configs/imx37_3stack_defconfig
+++ b/arch/arm/configs/imx37_3stack_defconfig
@@ -947,7 +947,7 @@ CONFIG_SND_VERBOSE_PROCFS=y
 #
 # ALSA ARM devices
 #
-# CONFIG_SND_MXC_SPDIF is not set
+CONFIG_SND_MXC_SPDIF=y
 
 #
 # SPI devices
diff --git a/sound/arm/mxc-alsa-spdif.c b/sound/arm/mxc-alsa-spdif.c
index 0ba8845..13ab28d 100644
--- a/sound/arm/mxc-alsa-spdif.c
+++ b/sound/arm/mxc-alsa-spdif.c
@@ -264,7 +264,7 @@ static int spdif_tx_init(void)
 {
 	unsigned long value;
 
-	value = 0x20514;
+	value = 0x20414;
 	__raw_writel(value, SPDIF_REG_SCR + spdif_base_addr);
 
 	value = 0xc0000;
-- 
1.5.4.4

